File: 2RingEmployeesMgmtFE/2ring-employees-mgmt/src/models/Employee.ts
import Position from "@/models/Position";
import PositionHistory from "@/models/PositionHistory";

export default class Employee {
  Id: number = 0;
  FirstName: string = "";
  Surname: string = "";
  Address: string = "";
  Birthday: Date | null = null;
  StartDate: Date = new Date();
  Salary: number = 0;
  IsArchived: boolean = false;
  Position: Position = new Position();
  LastModified: Date | any = new Date();
  PositionHistory: PositionHistory[] = [];

  constructor(obj?: any) {
    if (!obj) return;

    this.Id = obj.Id;
    this.FirstName = obj.FirstName;
    this.Surname = obj.Surname;
    this.Address = obj.Address;
    this.Birthday = new Date(obj.Birthday);
    this.StartDate = new Date(obj.StartDate);
    this.Salary = obj.Salary;
    this.IsArchived = obj.IsArchived;
    this.Position = new Position(obj.Position);
    this.LastModified = new Date(obj.LastModified);
    this.PositionHistory = <PositionHistory[]>obj.PositionHistory;
  }
}

File: 2RingEmployeesMgmtBE/Domain.Models/Employee.cs
using Domain.Models.Interfaces;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Domain.Models
{
    public class Employee : ITrackLastModified
    {
        public int Id { get; set; }
        public string FirstName { get; set; }
        public string Surname { get; set; }
        public string Address { get; set; }
        public DateTime Birthday { get; set; }
        public DateTime StartDate { get; set; }
        public float Salary { get; set; }
        public bool IsArchived { get; set; }
        public DateTime LastModified { get; set; }

        public virtual Position Position { get; set; }
        public virtual ICollection<EmployeePositionHistory> PositionHistory { get; set; } = new HashSet<EmployeePositionHistory>();
    }
}